conf/bkenv.sh

```diff
@@ -60,6 +57,17 @@ for token in $("$JAVA_BIN" -version 2>&1 | grep 'version "'); do
     fi
 done
 
+# Garbage collection options
+BOOKIE_GC="${BOOKIE_GC:-${PULSAR_GC}}"
+if [ -z "$BOOKIE_GC" ]; then
+  BOOKIE_GC="-XX:+PerfDisableSharedMem -XX:+AlwaysPreTouch"
+  if [[ $JAVA_MAJOR_VERSION -ge 21 ]]; then
+    BOOKIE_GC="-XX:+UseZGC -XX:+ZGenerational ${BOOKIE_GC}"
```

Comment:
Do we know ZGenerational is a better option for Pulsar?

Reply:
One reason why it's better since it doesn't have the double memory mapping issue. 

We don't have benchmark results yet, but I don't see a reason why we couldn't promote it as default. Most deployment methods override the defaults so it is very unlikely to cause regressions.

Reply:
https://netflixtechblog.com/bending-pause-times-to-your-will-with-generational-zgc-256629c9386b

Reply:
It will be interesting to check this benchmark, but I guess that could be a separate task.